app/src/main/java/co/epitre/aelf_lectures/SectionBibleFragment.java

```diff
@@ -124,8 +127,30 @@ public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
             mWebView.restoreState(savedInstanceState);
         } else {
             // Load default page
-            mWebView.loadUrl(BASE_RES_URL + "index.html");
+            Log.d(TAG,"Loading webview, KEY_BIBLE_LAST_PAGE is " + settings.getString(KEY_BIBLE_LAST_PAGE,"null"));
+            mWebView.loadUrl(settings.getString(KEY_BIBLE_LAST_PAGE,(BASE_RES_URL + "index.html")));
         }
+
+        //Save last URL
+        // onPageFinished infos found on https://stackoverflow.com/a/6720004
+        mWebView.setWebViewClient(new WebViewClient() {
+            @Override
+            public void onPageFinished(WebView view, String url) {
+                super.onPageFinished(mWebView, url);
+                String last_url = mWebView.getUrl();
+                Log.d(TAG, "last_url is " + last_url);
+                //TODO: Save the value in a persistent storage, maybe sharedpreferences and use it when the webview is re-created.
+                SharedPreferences settings = getActivity().getPreferences(Context.MODE_PRIVATE);
```

Comment:
You retrieve a `SharedPreference` using a different method than in line [59](https://github.com/HackMyChurch/aelf-dailyreadings/pull/13/files#diff-c83a481a5a2835d697777ae7a847e12eR59).

So you read from `PreferenceManager.getDefaultSharedPreferences(getContext())` and you write to `getActivity().getPreferences(Context.MODE_PRIVATE)` thus it cannot work.